javascript/javascript3/week2/rewrite.js

```diff
@@ -0,0 +1,37 @@
+/* I really think I don't understand this exercise but try to 
+find a way by looking at the codes we wrote before and get some idea from my classmates 
+that what should I do, but still I think I did it
+without exactly know what I did!
+anyway, I dont have some questions to ask about that and I think just need to practice more
+*/
+function setTimeoutPromise(seconds) {
+    return new Promise((resolve) => {
+        setTimeout(() => {
+            resolve();
+        }, seconds*1000);
+    })
+}
+setTimeoutPromise(3)
+    .then(() => {
+        console.log('Called after 3 seconds');
+    });
+
+function getCurrentLocation() {
+    return new Promise((resolve, reject) => {
+        navigator.geolocation.getCurrentPosition(position => {
+            resolve(position);
+        }, noPosition => {
+            reject(noPosition);
+        })
+    })
+}
+    
+getCurrentLocation()
+    .then((position) => {
+        // called when the users position is found
+        console.log(position);
+    })
+    .catch((error) => {
+        // called if there was an error getting the users location
+        console.log(error);
+    });
```

Comment:
Well you did it 🥇 



The purpose was the change the `navigator.geolocation.getCurrentPosition` to use a `Promise` instead of callbacks.
To do this, you wrap it in a `Promise`. 

And then give it a `onSuccess` and a `onError` callback function, that `resolve` and `reject` the Promise.

Which you did by wrapping `navigator.geolocation.getCurrentPosition` in the `Promise`
```
function getCurrentLocation() {
    return new Promise((resolve, reject) => {
        navigator.geolocation.getCurrentPosition(position => {
            resolve(position);
        }, noPosition => {
            reject(noPosition);
        })
    })
}
```

It makes it possible to use it like you did here:
```
getCurrentLocation()
    .then((position) => {
        // called when the users position is found
        console.log(position);
    })
    .catch((error) => {
        // called if there was an error getting the users location
        console.log(error);
    }); 
```

With a `Promise`, where we can use `.then` and `.catch` to handle the response.